src/tree/private_tree.rs

```diff
@@ -126,17 +150,13 @@ impl PrivateTree {
     /// ```
     ///
     /// Note that this overrides the `path_secrets`.
-    pub(crate) fn generate_path_secrets(
+    pub(crate) fn continue_path_secrets(
```

Comment:
The two functions are different semantically, despite some code overlap. Combining them broke the logic because some parts went missing in that process. It's much more readable and easier to audit that way.

Reply:
Ok, if you want to separate functions then factor out the common code and let both functions call this function to avoid code duplication.

Reply:
Done